imports/plugins/core/catalog/server/methods/catalog.js

```diff
@@ -816,7 +814,7 @@ Meteor.methods({
         // Should be a call similar to the line below.
         [field]: createHandle(Reaction.getSlug(value), _id) // handle should be unique
       };
-    } else if (field === "title" && doc.handle === doc._id) {
```

Comment:
The purpose of this is to ensure that permalink isn't updated every time someone changes a title. With the removal of the `ID` being the default permalink, this is definitely less useful; however, there should be new check added for the existence of the permalink and block updates if it's set.

The permalink changing every time the title changes will cause broken links. If someone wants to break their permalink, they can do so from the field itself, and make whatever redirects they want in their storefront if necessary.

What I'd like to see from this is:

- The permalink should be set the first time automatically based on the title. 
- The permalink doesn't update automatically with the title change if it's already set.
- The permalink is manually updateable through the field, which it is now.

Reply:
Hey @mikemurray , thanks for the feedback. One clarification about the first bullet point. 

When we create a product the title is null. I'll have the permalink updated automatically only the first time the title is set by the user. Until then it will be null (and after the first update it will not be updated automatically again). Sounds good or should we populate it initially with a default value?

Reply:
The permalink could be blank until the first update of the title. Products can't be published with a null title. Not sure about a null permalink, but the null title will make sure the permalink it set.

Reply:
is this thread resolved? asking for a friend that needs to test this.

Reply:
@manueldelreal No it's not! I'll ping you once it's ready for testing.

Reply:
I changed this locally and this is the behavior I see:

- Go to Products and create a new product. The title is not set yet and therefore the permalink is empty as well.

![image](https://user-images.githubusercontent.com/11715799/61210932-b8962200-a706-11e9-824d-e0b66a5ba6d7.png)

- Set a value for title -> permalink **is** updated automatically

![image](https://user-images.githubusercontent.com/11715799/61210963-dbc0d180-a706-11e9-99b7-ddc3dc2674cd.png)

- Update the title -> permalink **is not** updated automatically

![image](https://user-images.githubusercontent.com/11715799/61211034-0dd23380-a707-11e9-9536-982485179e05.png)

- Update the permalink manually -> update successful

![image](https://user-images.githubusercontent.com/11715799/61211065-25112100-a707-11e9-91c8-f46fe91cda0f.png)

However, if I delete the permalink manually and try to Publish, the system lets me do it. Should I add the same validation we have for Title and not allow them to publish if the permalink is null?
@mikemurray @zenweasel

Reply:
Actually the title validation is not on Publish, it doesn't let you delete to start with:

![image](https://user-images.githubusercontent.com/11715799/61212287-7bcc2a00-a70a-11e9-9e13-58f6ab1dc300.png)

Add the same for permalink?

Reply:
I think it's fine to make the permalink required since it no-longer defaults to the product id.

Reply:
@mikemurray Can you please review the changes I just pushed? Behaviour is the same as I described in my previous comment, plus if you try to delete the permalink you get this error:

![image](https://user-images.githubusercontent.com/11715799/61443783-d952b780-a952-11e9-9050-8a94afedba2d.png)